### PR TITLE
PD-333: Fix new glyphs not having fill prop applied correctly

### PR DESCRIPTION
**Pull request by DesignerDave**
## ✍️ Proposed changes

This PR sets the fill for the newer glyphs to `#000000` and adds tests to verify that this is the case with all new glyphs in the future. In the future, we should find a way to do this more resiliently (such as an implementation with `currentColor`, but this should make us less error-prone in the meantime.

🎟 _Jira ticket:_ [Fix new glyphs not having fill prop applied correctly](https://jira.mongodb.org/browse/PD-333)

## 🛠 Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] ~I have added necessary documentation (if appropriate)~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -118,5 +118,5 @@
     "packages": [
       "packages/*"
     ]
-  }
+  }
```
Comment (hswolff) [inline package.json]: y u add white space

File: packages/icon/src/Icon.spec.tsx [hunk 2]
```diff
@@ -1,20 +1,71 @@
...
 import { typeIs } from '@leafygreen-ui/lib';
 import { createIconComponent } from '.';
 
 afterAll(cleanup);
 
+describe('packages/Icon/glyphs/', () => {
+  const glyphPaths = fs
+    .readdirSync(path.resolve(__dirname, './glyphs'))
+    /**
+     * './glyphs/' contains an index.ts file, so we filter out
```
Comment (hswolff) [inline packages/icon/src/Icon.spec.tsx]: Can we also match that this list aligns with the exported object given from `index.ts`?

File: packages/icon/src/Icon.spec.tsx [hunk 3]
```diff
@@ -1,20 +1,71 @@
...
+     * anything that's not an svg.
+     */
+    .filter(path => /.*\.svg$/.test(path));
+
+  glyphPaths.forEach(glyphPath => {
+    describe(`${glyphPath}`, () => {
+      const { svg } = require(path.resolve(__dirname, `./glyphs/${glyphPath}`));
+
+      if (!svg) {
+        return;
```
Comment (hswolff) [inline packages/icon/src/Icon.spec.tsx]: Should we fail instead? This shouldn't ever be the case. `expect(svg).not.toBeNull()`?

File: packages/icon/src/Icon.spec.tsx [hunk 4]
```diff
@@ -1,20 +1,71 @@
...
+        Object.keys(obj).forEach(key => {
+          const currentValue = obj[key];
+
+          if (typeof currentValue === 'object') {
+            processGlyphObject(currentValue);
+          } else {
+            if (key === 'fill') {
+              const validFills = ['#000', '#000000', 'none'];
+
+              if (!validFills.includes(currentValue)) {
```
Comment (hswolff) [inline packages/icon/src/Icon.spec.tsx]: Could this not be the test itself? Such that if it fails the test shows what value it currently is? Maybe it doesn't matter?

File: package.json (hunk 1 above)
Comment (DesignerDave) [reply package.json]: Hmm our format script should've fixed this...

File: packages/icon/src/Icon.spec.tsx (hunk 3 above)
Comment (DesignerDave) [reply packages/icon/src/Icon.spec.tsx]: Actually, I think we can get rid of this altogether. We already explicitly filter out non-svg files from `glyphPaths`. I'm just gonna remove this.

File: packages/icon/src/Icon.spec.tsx (hunk 4 above)
Comment (DesignerDave) [reply packages/icon/src/Icon.spec.tsx]: I did something that _almost_ does what you're referring to. Unfortunately you won't see the current value, as I couldn't find a way to do something like `expect(currentValue).toBeWithin(validFills)`. If you have a better idea, let me know, but the new implementation is an improvement.

File: packages/icon/src/Icon.spec.tsx (hunk 2 above)
Comment (DesignerDave) [reply packages/icon/src/Icon.spec.tsx]: Done!

**Pull request merged**